cmake/pcl_targets.cmake

```diff
@@ -302,11 +302,7 @@ function(PCL_ADD_EXECUTABLE _name)
   endif()
   PCL_ADD_VERSION_INFO(${_name})
   # must link explicitly against boost.
-  if(UNIX AND NOT ANDROID)
-    target_link_libraries(${_name} ${Boost_LIBRARIES} pthread m ${CLANG_LIBRARIES})
```

Comment:
Can you explain the reason for deleting this special check? From what I can tell, it's targeting Unix builds done with clang. `${CLANG_LIBRARIES}` is set to `stdc++` in those cases. I just fear you might be breaking some runtime behavior by building against libc++ on Unix. Also please comment on the removaI of `libm`.

Reply:
My test system is using Ubuntu + Clang 6 and I can still compile everything with it. I believe `stdc++` is not necessary, because we are already adding a flag for C++14. Furthermore: As far as I can read you need `libm` in case your are including `math.h`, but replaced `math.h` already by `cmath`, so it is not necessary anymore.

Reply:
Can you try running a single executable, in this case an example, tool or app, just to validate? I had some issues with this mix of libc++ and libstdc++ before on mac and just want to make sure it's not gonna happen here.

Reply:
I started `pcl_multiscale_feature_persistence_example` with a random point cloud and this app is starting without any error message.